lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,181 @@
+        foreach ($disk in $disks) {
+            $disk_results = @{}
+            $pdisk = Get-PhysicalDisk -ErrorAction SilentlyContinue | Where-Object {
+                $_.DeviceId -eq $disk.Number
+            }
+            if ($pdisk) {
+                $disk_results["physical_disk"] += @{
+                    #media_type = $pdisk.MediaType
+                    size = "$($pSize = "{0:N3}" -f ($pdisk.Size / 1GB))$($pSize)gb"
+                    allocated_size = "$($pAllocSize = "{0:N3}" -f ($pdisk.AllocatedSize / 1GB))$($pAllocSize)gb"
+                    device_id = $pdisk.DeviceId
+                    friendly_name = $pdisk.FriendlyName
+                    operational_status = $pdisk.OperationalStatus
+                    health_status = $pdisk.HealthStatus
+                    bus_type = $pdisk.BusType
+                    usage_type = $pdisk.Usage
+                    supported_usages = $pdisk.SupportedUsages
+                    spindle_speed = "$($pdisk.SpindleSpeed)rpm"
+                    firmware_version = $pdisk.FirmwareVersion
+                    physical_location = $pdisk.PhysicalLocation
+                    manufacturer = $pdisk.Manufacturer
+                    model = $pdisk.Model
+                    can_pool = $pdisk.CanPool
+                    indication_enabled = "$([string]$pdisk.IsIndicationEnabled)"
+                    partial = $pdisk.IsPartial
+                    serial_number = $pdisk.SerialNumber
+                    object_id = $pdisk.ObjectId
+                    unique_id = $pdisk.UniqueId
+                }
+                if (-not $pdisk.CanPool) {
+                    $disk_results.physical_disk.cannot_pool_reason = $pdisk.CannotPoolReason
+                }   
+                $vdisk = Get-VirtualDisk -PhysicalDisk $pdisk -ErrorAction SilentlyContinue
+                if ($vdisk) {
+                    $disk_results["virtual_disk"] += @{
+                        size = "$($vDSize = "{0:N3}" -f ($vdisk.Size / 1GB))$($vDSize)gb"
+                        allocated_size = "$($vDAllocSize = "{0:N3}" -f ($vdisk.AllocatedSize / 1GB))$($vDAllocSize)gb"
+                        footprint_on_pool = "$($vDPrint = "{0:N3}" -f ($vdisk.FootprintOnPool / 1GB))$($vDPrint)gb"
+                        name = "$([string]$vdisk.name)"
+                        friendly_name = $vdisk.FriendlyName
+                        operational_status = $vdisk.OperationalStatus
+                        health_status = $vdisk.HealthStatus
+                        provisioning_type = $vdisk.ProvisioningType
+                        allocation_unit_size = "$($vdisk.AllocationUnitSize / 1KB)kb"
+                        media_type = $vdisk.MediaType
+                        parity_layout = "$([string]$vdisk.ParityLayout)"
+                        access = $vdisk.Access
+                        detached_reason = $vdisk.DetachedReason
+                        write_cache_size = "$($vdisk.WriteCacheSize)byte"
```

Comment:
s/byte/bytes/ but this depends on whether we want to include the units on the values.

Reply:
I have implemented that, but would be removed depending on above decision.